# Palmer Penguins

***
[Penguins](https://allisonhorst.github.io/palmerpenguins/reference/figures/lter_penguins.png) 

This notebook contains my analysis of the famous palmer penguins dataset.

The dataset is available [ on GitHub](https://allisonhorst.github.io/palmerpenguins/).

## Imports

***

We use pandas for the DataFrame data structure.

It allows us to investigate CSV files, amongst other features.

In [1]:
# Data frames
import pandas as pd

## Load Data

Load the palmer penguins data set from a URL

In [2]:
# Loads the penguin dataser
df= pd.read_csv('https://raw.githubusercontent.com/mwaskom/seaborn-data/master/penguins.csv')

The data is now loaded and we can inspect it

In [3]:
# Let's have a look
df

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
339  Gentoo     Biscoe             NaN            NaN                NaN   
340  Gentoo     Biscoe            46.8           14.3              215.0   
341  Gentoo     Biscoe            50.4           15.7              222.0   
342  Gentoo     Biscoe            45.2           14.8              212.0   
343  Gentoo     Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  
0         3750.0    MALE  
1         3800.0  FEMALE  
2         3250.0  FEMALE  
3            NaN     NaN  
4         3450.0  FEMALE  
..           ...     ...  
339          NaN     NaN  
340       4850.0  FEMALE  
341       5750.0    MALE  
342       5200.0  FEMALE  
343       5400.0    MALE  

[344 rows x 7 columns]

Inspecting Data

In [4]:
# look at the first row
df.iloc[0]

species                 Adelie
island               Torgersen
bill_length_mm            39.1
bill_depth_mm             18.7
flipper_length_mm        181.0
body_mass_g             3750.0
sex                       MALE
Name: 0, dtype: object

In [5]:
# Sex of penguins
df['sex']

0        MALE
1      FEMALE
2      FEMALE
3         NaN
4      FEMALE
        ...  
339       NaN
340    FEMALE
341      MALE
342    FEMALE
343      MALE
Name: sex, Length: 344, dtype: object

In [6]:
# count the number of penguins of each sex
df['sex'].value_counts()

sex
MALE      168
FEMALE    165
Name: count, dtype: int64

In [7]:
#describe the data set
df.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
count      342.000000     342.000000         342.000000   342.000000
mean        43.921930      17.151170         200.915205  4201.754386
std          5.459584       1.974793          14.061714   801.954536
min         32.100000      13.100000         172.000000  2700.000000
25%         39.225000      15.600000         190.000000  3550.000000
50%         44.450000      17.300000         197.000000  4050.000000
75%         48.500000      18.700000         213.000000  4750.000000
max         59.600000      21.500000         231.000000  6300.000000

## Tables

***

| Species  |Bill Length (mm)|Body Mass (g)|
|--------- |----------------:|-------------:|
|Adelie    |            38.8|         3701|
|Chinstrap |            48.8|         3733|
|Gentoo    |            47.5|         5076|


# Presentation of the dataset

The Palmer Penguins dataset is a popular dataset used for data analysis and machine learning tasks. It contains measurements and other attributes of penguins collected from three species: Adelie, Chinstrap, and Gentoo, on three islands in the Palmer Archipelago, Antarctica: Dream, Torgersen and Biscoe.

Here is the descriptive statistics explained in the data set:


**count:** This tells you how many data points you have for each measurement. In this case, you have 342 data points for all four measurements.

**mean:** This is the average value for each measurement. For example, the average bill length is approximately 43.92 mm, the average bill depth is approximately 17.15 mm, the average flipper length is approximately 200.92 mm, and the average body mass is approximately 4201.75 g.

**std:** This is the standard deviation, which measures the amount of variation or dispersion in the data set. It tells you how spread out the values are from the mean. For instance, the standard deviation for bill length is approximately 5.46 mm, for bill depth is approximately 1.97 mm, for flipper length is approximately 14.06 mm, and for body mass is approximately 801.95 g.

**min:** This is the minimum value observed in the data set. For example, the minimum bill length observed is 32.1 mm, the minimum bill depth observed is 13.1 mm, the minimum flipper length observed is 172 mm, and the minimum body mass observed is 2700 g.

**25% (Q1):** This is the first quartile, which represents the value below which 25% of the data fall. For example, 25% of the penguins have a bill length of 39.225 mm, a bill depth of 15.6 mm, a flipper length of 190 mm, and a body mass of 3550 g.

**50% (Q2 or median):** This is the median value, which represents the middle value of the data set when it is ordered from smallest to largest. For example, the median bill length is 44.45 mm, the median bill depth is 17.3 mm, the median flipper length is 197 mm, and the median body mass is 4050 g.

**75% (Q3):** This is the third quartile, which represents the value below which 75% of the data fall. For example, 75% of the penguins have a bill length of 48.5 mm, a bill depth of 18.7 mm, a flipper length of 213 mm, and a body mass of 4750 g.

**max:** This is the maximum value observed in the data set. For example, the maximum bill length observed is 59.6 mm, the maximum bill depth observed is 21.5 mm, the maximum flipper length observed is 231 mm, and the maximum body mass observed is 6300 g.
These statistics provide a comprehensive overview of the distribution and central tendencies of the data for each measurement.

I'm going now to get variable type of our data set and explaining it:

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 7 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     342 non-null    float64
 3   bill_depth_mm      342 non-null    float64
 4   flipper_length_mm  342 non-null    float64
 5   body_mass_g        342 non-null    float64
 6   sex                333 non-null    object 
dtypes: float64(4), object(3)
memory usage: 18.9+ KB


#### Overview of the variables:

**species:** The species of penguin. It can be one of three values: "Adelie", "Chinstrap", or "Gentoo".  
**island:** The island where the penguin was observed. It can be one of three values: "Biscoe", "Dream", or "Torgersen".  
**bill_length_mm:** The length of the penguin's bill in millimeters.  
**bill_depth_mm:** The depth of the penguin's bill in millimeters.  
**flipper_length_mm:** The length of the penguin's flipper in millimeters.  
**body_mass_g:** The body mass of the penguin in grams.  
**sex:** The gender of the penguin. It can be one of three values: "male", "female", or "NA" (indicating unknown).  
year: The year in which the data was collected.  


### Notes:
The dataset may contain missing values (indicated by "NA" or NaN).
The measurements (bill length, bill depth, flipper length, body mass) are numeric variables representing physical characteristics of the penguins.
The categorical variables (species, island, sex) provide information about the species, habitat, and gender of the penguins.
The dataset is often used for classification tasks (e.g., predicting the species of a penguin based on its physical characteristics) and exploratory data analysis.



### Modeling the variables

To model the variables in the Palmer Penguins dataset in Python, I have considered the following:

- Categorical Variables:  
  - Species: This variable represents different penguin species (Adelie, Chinstrap, or Gentoo). It should be treated as a categorical variable because each species represents a distinct category with no inherent ordering.  
  - Island: The island where the penguin was observed (Dream, Torgersen, or Biscoe). Like species, this variable should also be treated as categorical since the islands represent distinct categories without any natural ordering.  
  - Sex: The sex of the penguin (male, female, or NA if unknown). This is also a categorical variable since each sex represents a distinct category.
- Numeric Variables:   
  - Bill Length (bill_length_mm): This represents the length of the penguin's bill in millimeters. It is a numeric variable that can be used for modeling, such as regression analysis.  
  - Bill Depth (bill_depth_mm): Similar to bill length, this represents the depth of the penguin's bill in millimeters and can also be treated as a numeric variable.  
  - Flipper Length (flipper_length_mm): The length of the penguin's flipper in millimeters. This is another numeric variable suitable for modeling.
  - Body Mass (body_mass_g): The body mass of the penguin in grams. This numeric variable provides information about the weight of the penguins and can be useful for modeling purposes.  
- Target Variable (Optional):  
  - Depending on your analysis goals, you may choose to define a target variable for predictive modeling tasks. For example, if you want to predict the species of a penguin based on its physical characteristics, then species can be considered as the target variable for classification tasks.  


Treating categorical variables as categorical types ensures that Python understands them as distinct categories rather than treating them as numeric values with some order or significance.  I'm moving a string variable to a categorical one so that Phyton can analyze it.
Numeric variables for modeling allow to perform mathematical operations and statistical analyses on these variables. This enables tasks like correlation analysis, and visualization.  
The choice of variables was made on the objective of the analysis. In my example I have done so to understand the relationship between penguin physical characteristics.

***

### End